integrations/access/accessrequest/app.go

```diff
@@ -345,6 +415,14 @@ func (a *App) getMessageRecipients(ctx context.Context, req types.AccessRequest)
 	// This can happen if this set contains the channel `C` and the email for channel `C`.
 	recipientSet := common.NewRecipientSet()
 
+	recipients := a.recipientsFromAccessMonitoringRules(ctx, req)
+	for _, recipient := range recipients.ToSlice() {
```

Comment:
Consider adding a `.ForEach` to `RecipientSet` so that you don't have to copy all the data into a slice just to iterate over it.